page-objects/pages/TASKListPage.js

```diff
@@ -160,67 +147,71 @@ async function clickDeleteTaskButtonInPopup() {
 
 async function clickAtTask(taskName) {
 	await waitHelpers.waitUntilNuxtClientLoads();
-	await driver.pause(5000);
-	let clickOnThatTask = (await getTaskFromNuxtClient(taskName)).toString();
-	await elementHelpers.scrollToElement(clickOnThatTask);
-	await elementHelpers.clickAndWait(clickOnThatTask);
+	let clickOnThatTask = (await getTaskFromTaskOverview(taskName));
+	if (clickOnThatTask === undefined){
+		let isTaskClickable = new Boolean(false);
+		while (!isTaskClickable){
+			elementHelpers.moveToElement(clickOnThatTask);
+			await waitHelpers.waitUntilElementIsVisible(clickOnThatTask);
+			if (clickOnThatTask.isDisplayedInViewport()){
+				isTaskClickable = true;
+			}
+		}
+	}
+	if (clickOnThatTask.isDisplayedInViewport()) {
+		await elementHelpers.clickAndWait(clickOnThatTask);
+	}else{
+		await elementHelpers.moveToElement(clickOnThatTask);
+		await elementHelpers.clickAndWait(clickOnThatTask);
+	}
 }
 
-async function getTaskFromNuxtClient(taskName) {
-	await driver.pause(3000);
-	await waitHelpers.waitUntilElementIsVisible(taskOverviewLoad);
-	await driver.pause(3000);
-	const taskOverviewResult = await getNuxtTaskList();
-	const taskIndex = taskOverviewResult.indexOf(taskName);
-	let clickOnTask = taskOverviewResult[taskIndex];
-	const taskInTheList = '//div[text() =' + "'" + clickOnTask + "'" + ']';
-	return taskInTheList;
+async function getTaskFromTaskOverview(taskName) {
+	let tasksOnThePage = [];
+	await waitHelpers.waitUntilNuxtClientLoads();
+	await waitHelpers.waitUntilElementIsPresent(taskSection);
+	await driver.$(taskSection).$$(taskTitle).forEach(async function (element)  {
+		tasksOnThePage.push(await element.getText());
+	})
+	let isTaskInTheList = new Boolean(false);
+	isTaskInTheList =  (tasksOnThePage.includes(taskName)) ? driver.$(mod_extsprintf.sprintf(taskTitleText, taskName)) : false ;
+	return isTaskInTheList
 }
 
 async function taskDisplayed(taskName) {
-	let taskInTheList = (await getTaskFromNuxtClient(taskName)).toString();
+	let taskInTheList = (await getTaskFromTaskOverview(taskName));
```

Comment:
I'm wondering if we should add 'expect(taskInTheList).to.equal(true);' below await waitHelpers.waitUntilElementIsPresent(taskInTheList);?
Of course, if the element is present, it's there. But to have a full check and to be align with the opposite check, we could add the expect. What do you think?

Reply:
Its done.